File: sort/冒泡排序/bubble.go
package main

import (
	"fmt"
)

func main() {
	number := []int{95, 45, 15, 78, 84, 51, 24, 12}
	fmt.Println("Sort Result:", BubbleSort(number))
	fmt.Println("Sort Result:", BubbleSort1(number))
}

/*
	BubbleSort 冒泡排序
	input: [95 45 15 78 84 51 24 12]
	output: [12 15 24 45 51 78 84 95]
*/
func BubbleSort(values []int) []int {
	flag := true
	vLen := len(values)
	for i := 0; i < vLen-1; i++ {
		flag = true
		for j := 0; j < vLen-i-1; j++ {
			if values[j] > values[j+1] {
				values[j], values[j+1] = values[j+1], values[j]
				flag = false
				continue
			}
		}
		if flag {
			break
		}
	}
	return values
}

func BubbleSort1(values []int) []int {
	length := len(values)
	for i := 0; i < length; i++ {
		for j := length - 1; j > i; j-- {
			if values[j] < values[j-1] {
				tmp := values[j-1]
				values[j-1] = values[j]
				values[j] = tmp
			}
		}
	}
	return values
}

File: sort/冒泡排序/bubble.java
/* 
    input:
        4
        1
        5
        2
        3
    output:
        1
        2
        3
        4
        5
*/
public class bubble {
    public static void main(String []args){
        int[] array = {4,1,5,2,3};
        System.out.println("input:");
        for(int a:array)
            System.out.println(a);
        System.out.println("output:");
        for(int b:bubbleSort(array))
            System.out.println(b);
    }
    // 排序
    public static int[] bubbleSort(int []arr){
        for(int i =0;i<arr.length-1;i++) { 
            for(int j=0;j<arr.length-1;j++) {
                if(arr[j]>arr[j+1]) {
                    int temp = arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=temp;
                }
            }    
        }
        return arr;
    }
}

